src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala

```diff
@@ -13,11 +14,14 @@ trait WorkerMain[S] {
 
   protected[this] def init(args: Option[Array[String]]): S
 
-  protected[this] def work(ctx: S, args: Array[String], out: PrintStream): Unit
+  protected[this] def work(ctx: S, args: Array[String], out: PrintStream, workDir: Path): Unit
```

Comment:
I thought about that, but went with a `Path` instead because:
 * Every work request has a work directory. It's either this directory `""` or a specific sandbox directory
 * You just move the weirdness to the other side with `.getOrElse("")` in many places..

Reply:
Is `""` even a valid path though?

Reply:
Yeah, it ends up being the current working directory of the process, AFAIK. Which for a non-sandboxed worker is the correct working directory.

```scala> Paths.get("").toAbsolutePath().toString()
val res3: String = /home/james
```